### PR TITLE
obsolete impossible gunmods

### PR DESCRIPTION
**Pull request by ymber**
Removes some more gunmods that aren't mechanically viable. I left some in that could maybe kinda work if you squint really hard. While #24479 is open I didn't touch ones that could exist but have weird numbers.

### CODE REVIEW

**Comment by Firestorm01X2** [comment]
I ll rewert my gunmod chahed. So do not worry

**Comment by cainiaowu** [comment]
You dont move them into obsolete.json, itself is a obsoleted method.
Just remove the definition and use migration.json to transform saved item into another item instead.

**Comment by Firestorm01X2** [comment]
Feel free to change everything.

**Comment by ymber** [comment]
@cainiaowu There's nothing to transform them too. I just removed them from spawn lists and marked them for removal so they can be phased out.

**Comment by Firestorm01X2** [comment]
Done. I've reverted my changes to mods. Fell free to rebalance mods.
I recommend to set dispersion from -45 or -60 to -15.

Hovewer Keving recommend to use 5 that is basically nullify effect of accuracy bonus. 15 should be small arleady.

**Comment by Firestorm01X2** [comment]
@ymber 
> I didn't touch ones that could exist but have weird numbers.

I going to do it now?

**Comment by ymber** [comment]
@Firestorm01X2 You handle the numbers for existing mods in #24479. I'm just removing the most impractical mods in this PR. I've got some ideas for changes to existing ones but they'd have to come after changes to how attachments and part replacements work so that's longer term.

**Comment by Mecares** [comment]
Maybe you could move them to the "more craftable guns" mod instead of removing them? Modding your guns is fun.

**Comment by plazma-rush** [comment]
Does the RAPIDFIRE flag and its effects actually have to be removed? I could see it being used for energy weapon gunmods, or weapon-specific methods of increasing fire rate (Granted, the 1.5x multiplier would still be way too high and a 1.25 or 1.3 would be more realistic, but that's outside of the scope of this PR).

**Comment by Firestorm01X2** [comment]
This should be refernced:
https://github.com/CleverRaven/Cataclysm-DDA/issues/24429

**Comment by ymber** [comment]
@plazma-rush If you've got something in mind you want to do with it I'll leave it in. Otherwise I think it's likely to get lost in the code and never used again.

**Comment by Rivet-the-Zombie** [comment]
What's wrong with rifled shotgun barrels? Those are a real thing.

**Comment by plazma-rush** [comment]
Yeah, feel free to remove it. I've got some ideas but nothing very specific.

**Comment by Firestorm01X2** [comment]
Oh, guys. You don't see full picture here. Soon all accuracy mods will be nerfed to be useless.

File: data/json/npcs/items_generic.json [hunk 1]
```diff
@@ -181,10 +181,7 @@
       ["bacon", 2],
       ["banana", 1],
       ["bandages", 25],
-      ["barrel_big", 1],
       ["barrel_ported", 1],
-      ["barrel_rifled", 1],
-      ["barrel_small", 1],
```
Comment (Night-Pryanik) [inline data/json/npcs/items_generic.json]: Small barrel is removed from the item group, but doesn't moved to the `obsolete.json`?

**Comment by Firestorm01X2** [comment]
Also. There are Leadworks version of barrels thet relies on obsolated gunmods
lead_barrel_big
lead_barrel_heavy_duty

**Pull request merged**